wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -1496,6 +1502,49 @@ impl<V> MutableTrie<V> {
         }
     }
 
+    pub fn delete_iter(
+        &mut self,
+        loader: &mut impl FlatLoadable,
+        iterator: &mut Iterator,
+    ) -> Option<()> {
+        let mut key_iter = iterator.get_key().iter();
```

Comment:
I don't think this is accurate since
```
    /// Key at the current position of the iterator.
    pub(crate) key:        Vec<u8>,
```

Reply:
Shouldn't we be able to simply remember the node at which the iterator was rooted? And avoid this traversal altogether.